s3/main.tf

```diff
@@ -149,6 +161,223 @@ resource "aws_s3_bucket_public_access_block" "public_access_block" {
+  validation_method = "DNS"
+
+  lifecycle {
+    create_before_destroy = true
+  }
+
+  tags = local.tags
+}
+
+data "aws_route53_zone" "selected" {
+  count        = var.config.serve_static ? 1 : 0
+  provider     = aws.domain-cdn
+  name         = "${var.application}.uktrade.digital"
+  private_zone = false
+}
+
+resource "aws_route53_record" "cert_validation" {
+  count    = var.config.serve_static ? 1 : 0
+  provider = aws.domain-cdn
+
+  name            = element(aws_acm_certificate.certificate[0].domain_validation_options[*].resource_record_name, 0)
+  type            = element(aws_acm_certificate.certificate[0].domain_validation_options[*].resource_record_type, 0)
+  zone_id         = data.aws_route53_zone.selected[0].id
+  records         = [element(aws_acm_certificate.certificate[0].domain_validation_options[*].resource_record_value, 0)]
+  ttl             = 60
+  depends_on      = [aws_acm_certificate.certificate]
+  allow_overwrite = true
+}
+
+resource "aws_acm_certificate_validation" "certificate_validation" {
+  count                   = var.config.serve_static ? 1 : 0
+  provider                = aws.domain-cdn
+  certificate_arn         = aws_acm_certificate.certificate[0].arn
+  validation_record_fqdns = [aws_route53_record.cert_validation[0].fqdn]
+  depends_on              = [aws_route53_record.cert_validation]
+}
+
+resource "aws_route53_record" "cloudfront_domain" {
+  count    = var.config.serve_static ? 1 : 0
+  provider = aws.domain-cdn
+  name     = aws_s3_bucket.this.bucket
+  type     = "A"
+  zone_id  = data.aws_route53_zone.selected[0].id
+  alias {
+    name                   = aws_cloudfront_distribution.s3_distribution[0].domain_name
+    zone_id                = aws_cloudfront_distribution.s3_distribution[0].hosted_zone_id
+    evaluate_target_health = false
+  }
+}
+
+
+resource "aws_cloudfront_origin_request_policy" "forward_content_type" {
+  count    = var.config.serve_static ? 1 : 0
+  provider = aws.domain-cdn
+  name     = "ForwardContentTypePolicy"
+
+  headers_config {
+    header_behavior = "whitelist"
+    headers {
+      items = ["Content-Type"]
+    }
+  }
+
+  query_strings_config {
+    query_string_behavior = "none"
+  }
+
+  cookies_config {
+    cookie_behavior = "none"
+  }
+}
+
+data "aws_cloudfront_cache_policy" "example" {
+  name = "Managed-CachingOptimized"
+}
+
+resource "aws_cloudfront_distribution" "s3_distribution" {
+  # checkov:skip=CKV2_AWS_32: Ensure CloudFront distribution has a response headers policy attached
+  # checkov:skip=CKV2_AWS_47: Ensure AWS CloudFront attached WAFv2 WebACL is configured with AMR for Log4j Vulnerability
+  # checkov:skip=CKV_AWS_68: CloudFront Distribution should have WAF enabled
+  # checkov:skip=CKV_AWS_86: Ensure CloudFront distribution has Access Logging enabled
+  # checkov:skip=CKV_AWS_305: Ensure CloudFront distribution has a default root object configured
```

Comment:
I don't think we want this, as we'll want individual service teams to configure whatever objects they want

Reply:
Correct, this needs to be available to the team to configure.